src/osd/scrubber/osd_scrub.h

```diff
@@ -98,29 +98,22 @@ class OsdScrub {
    *  locking: not using the jobs_lock
    */
   void update_job(
-      Scrub::ScrubJobRef sjob,
+      Scrub::ScrubJob& sjob,
       const Scrub::sched_params_t& suggested,
       bool reset_notbefore);
 
   /**
    * Add the scrub job to the list of jobs (i.e. list of PGs) to be periodically
    * scrubbed by the OSD.
-   * The registration is active as long as the PG exists and the OSD is its
-   * primary.
-   *
-   * See update_job() for the handling of the 'suggested' parameter.
-   *
-   * locking: might lock jobs_lock
    */
-  void register_with_osd(
-      Scrub::ScrubJobRef sjob,
-      const Scrub::sched_params_t& suggested);
+  void enqueue_target(Scrub::ScrubJob& sjob);
```

Comment:
This takes a mutable reference because the implementation updates target_queued --ScrubQueue::enqueue_target actually asserts !target_queued.  The comment here does not mention this behavior or that the caller is responsible for ensuring that target_queued is cleared.  This is particularly confusing as pg_scrubber appears to be responible for *clearing* this flag.  I'd really prefer that components outside of pg simply queue the ScrubJob and ignore any fields they don't specifically need to access for that purpose.  External interfaces like this one should take const references to ScrubJob.

Reply:
modified